### DISCUSSION TITLE
Carnot heat pump is not realistic

### DISCUSSION THREAD

**Discussion by JordiGarciaCespedes** [lbl-srg/modelica-buildings#3355] [Ideas]
I have been surfing through the water-to-water heat pumps in Buildings library, and I found there is an important remark to make about the model `Buildings.Fluid.HeatPumps.Carnot_TCon`. I do not understand why the outlet heat flow at the condenser grows with the condenser temperature. Ideally, a heat pump should show a decreasing heat capacity as the required Temperature at the condenser is rised.
![image](https://user-images.githubusercontent.com/105857531/233313490-7e1031e5-09f5-4126-8779-ad7631bfa669.png)
![image](https://user-images.githubusercontent.com/105857531/233311631-01347b3d-53b4-4d76-8728-3f06b837fd7f.png)
Apart from it, COP value shows unrealistic values when the heat pump is considered in OFF mode. I wish I knew more about Modelica to dig into the code of the inherited classes and solve it...
I see that alternative models like the `Buildings.Fluid.HeatPumps.ReciprocatingWaterToWater` performs in a more realistic way, although its defining parameters are not really useful. It is difficult to translate "Piston Displacement", "Superheating at evaporator", "Heat transfer coefficients at evaporator and condenser" (among others) into "nominal heat capacity" or "nominal mass flows". Any ideas to overcome this...?
Thank you

**Comment by mwetter** [comment]
The model `Buildings.Fluid.HeatPumps.Carnot_TCon` is expected to increase the condenser heat flow rate if the set point is increased (and mass flow and inlet temperatures are constant. From the documentation:

> The control input is the setpoint of the condenser leaving temperature, which is met exactly at steady state if the heat pump has sufficient capacity.

The COP is probably that high as it computed based on the fluid temperatures, which at initialization may be close together. See https://simulationresearch.lbl.gov/modelica/releases/v9.1.0/help/Buildings_Fluid_HeatPumps.html#Buildings.Fluid.HeatPumps.Carnot_TCon for the equation that is used. After startup you should get reasonable values.

For the `ReciprocatingWaterToWater` you would need to do a parameter estimation to determine the values that meet your product data.

**Reply by JordiGarciaCespedes** [reply]
Ok, then I understand this model is designed for a constrained scenario defined by specific values of T_evap_in /T_cond_out and m_flow_cond/m_flow_evap and nominal COP (maybe steady state, but not dynamic). I just would like to send the idea that this particular model seems easier to understand (conceptually speaking) and to implement by users, because it does not requires any catalogue data. However, it does not provide a representative response in a dynamic scenario, where the T of the source and the T of the load can change along the year (at least, it does not seem evident to me how to do it).

**Reply by mwetter** [reply]
I don't understand the concern in the last sentence: The actual COP is computed based on current temperatures on the hot and cold side.

**Reply by JordiGarciaCespedes** [reply]
Yes, the COP is computed on hot and cold side temperatures, but also the value of Q_cond. What I say is that it is not realistic that Q_cond (kW) grows as Temperature of the hot side increases (see plots in my first message). In a real heat water-to-water heat pump, Q_cond is reduced as the hot side temperature goes up.

**Reply by mwetter** [reply]
This model adds as much condenser heat as needed to meet the condenser leaving set point temperature TSet if it is below QCon_flow_max. This may be thought off as a heat pump with variable speed compressor.
